www/class/CentreonLDAPAdmin.class.php

```diff
@@ -567,26 +650,28 @@ public function getServersFromResId($arId)
      */
     private function manageContactPasswords($arId)
     {
-        $result = $this->db->query(
+        $result = $this->db->prepare(
             'SELECT ari_value ' .
             'FROM auth_ressource_info ' .
-            'WHERE ar_id = ' . $this->db->escape($arId) . ' ' .
-            'AND ari_name = "ldap_store_password" '
+            'WHERE ar_id = :ar_id AND ari_name = "ldap_store_password" '
         );
+        $result->bindValue(':ar_id', $arId, \PDO::PARAM_INT);
+        $result->execute();
         if ($row = $result->fetch()) {
             if ($row['ari_value'] == '0') {
-                $statement = $this->db->prepare("SELECT contact_id FROM contact WHERE ar_id = :arId");
-                $statement->bindValue(':arId', $arId, \PDO::PARAM_INT);
+                $statement = $this->db->prepare("SELECT contact_id FROM contact WHERE ar_id = :ar_id");
+                $statement->bindValue(':ar_id', $arId, \PDO::PARAM_INT);
                 $statement->execute();
                 $ldapContactIdList = [];
                 while ($row = $statement->fetch()) {
                     $ldapContactIdList[] = $row['contact_id'];
                 }
                 if (!empty($ldapContactIdList)) {
-                    $contactIds = implode(', ', $ldapContactIdList);
-                    $this->db->query(
-                        "DELETE FROM contact_password WHERE contact_id IN ($contactIds)"
+                    $statement = $this->db->prepare(
+                        "DELETE FROM contact_password WHERE contact_id IN ("
+                        . implode(',', $ldapContactIdList) . ")"
                     );
+                    $statement->execute();
```

Comment:
this change is not needed
You can keep `query` if you don't bind values

Reply:
done